extensions/amp-ad/0.1/test/test-amp-ad-api-handler.js

```diff
@@ -117,6 +117,41 @@ describe('amp-ad-api-handler', () => {
             });
       });
 
+      it('should resolve and resize on message "render-start" w/ size if '
+          + 'render-start is implemented by 3P', () => {
+        adImpl.adType = 'doubleclick';
+        sandbox.stub(adImpl, 'attemptChangeSize', (height, width) => {
+          expect(height).to.equal(217);
+          expect(width).to.equal(114);
+          return Promise.resolve();
+        });
+        apiHandler = new AmpAdApiHandler(adImpl, adImpl.element);
+        const beforeAttachedToDom = element => {
+          element.setAttribute('data-amp-3p-sentinel', 'amp3ptest' + testIndex);
+          startUpPromise = apiHandler.startUp(element, true);
+        };
+        return createIframeWithMessageStub(window, beforeAttachedToDom)
+            .then(newIframe => {
+              iframe = newIframe;
+              expect(iframe.style.visibility).to.equal('hidden');
+              iframe.postMessageToParent({
+                sentinel: 'amp3ptest' + testIndex,
+                type: 'render-start',
+                width: 114,
+                height: 217,
+              });
+              return startUpPromise.then(() => {
+                expect(iframe.style.visibility).to.equal('');
+                return iframe.expectMessageFromParent('amp-' + JSON.stringify({
+                  requestedWidth: 114,
+                  requestedHeight: 217,
+                  type: 'embed-size-changed',
```

Comment:
we can only sort the message above to match the message below. The message below will be converted to a string and order matters.

I changed the order of the message above since it's an object. Also made the change in other places.